src/main/java/core/basesyntax/service/Parser.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.fruit.Record;
+import java.util.List;
+import java.util.Map;
+
+public interface Parser {
+    List<Record> parseFileData(List<String> data);
```

Comment:
I think separating the interface into separate ones would be better. The LSP the principle tells us that several small interfaces are better than one large one. And you used it in different places, so maybe will be better create Interfase Parser to input data and ReportGenerator for output